raiden-ts/src/transport/epics.ts

```diff
@@ -433,7 +483,13 @@ export const initMatrixEpic = (
           }),
           switchMapTo(matrix$),
           delay(1e3), // wait 1s before starting matrix, so event listeners can be registered
-          mergeMap(matrix => matrix.startClient({ initialSyncLimit: 0 })),
+          withLatestFrom(config$),
+          mergeMap(async ([matrix, config]) =>
+            matrix.startClient({
+              initialSyncLimit: 0,
+              filter: await createFilter(config, server, matrix),
```

Comment:
Nitpick: createFilter and startClient are two separate async calls. Put each on different mergeMap so, in case of unsubscribe/complete between both, the second isn't unconditionally called as with plain composed async calls

Reply:
Thanks that is a really good idea :)